apps/api/src/app/workflows-v2/usecases/delete-workflow/delete-workflow.usecase.ts

```diff
@@ -0,0 +1,63 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+
+import { EnvironmentRepository, NotificationTemplateRepository } from '@novu/dal';
+import { DeleteWorkflowCommand } from './delete-workflow.command';
+import { ControlValuesRepository, MessageTemplateRepository } from '@novu/dal/src';
+import { WorkflowNotFoundException } from '../../exceptions/workflow-not-found';
+
+@Injectable()
+export class DeleteWorkflowUseCase {
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private environmentRepository: EnvironmentRepository,
+    private msgTemplateRepo: MessageTemplateRepository,
+    private controlValuesRepository: ControlValuesRepository
+  ) {}
+
+  async execute(command: DeleteWorkflowCommand): Promise<void> {
+    await this.validateEnvironment(command);
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Why is this check for existing of the environment necessary? I would expect that the environment already exists at this point of the API request, because the authentication & authorization checks were successful upstream.‏

Reply:
great question, was trying to follow an existing structure, will happily remove it from all use cases @djabarovgeorge WDYT ?

Reply:
IMO it is safe to remove it, usually if I fetch an entity i add such validation in order to let typescript know it is not nullable, however here we do not need additional data about the environment so the fetch can be removed.

Thanks for pointing it up 👍🏻  I will remove it in my commit.